lib/provider/camunda/element-templates/cmd/ChangeElementTemplateHandler.js

```diff
@@ -48,100 +48,174 @@ function ChangeElementTemplateHandler(modeling, commandStack, bpmnFactory) {
-
-      commandStack.execute('properties-panel.update-businessobject-list', {
-        element: element,
-        currentObject: extensionElements,
-        propertyName: 'values',
-        objectsToAdd: [ newProperties ],
-        objectsToRemove: oldProperties ? [ oldProperties ] : []
-      });
+      if (!context) {
+        context = getOrCreateExtensionElements(element);
+      }
+      if (!inside) {
+        oldProperties = findExtension(element, 'camunda:Properties');
+
+        commandStack.execute('properties-panel.update-businessobject-list', {
+          element: element,
+          currentObject: context,
+          propertyName: 'values',
+          objectsToAdd: [ newProperties ],
+          objectsToRemove: oldProperties ? [ oldProperties ] : []
+        });
+      }
+      else {
+        commandStack.execute('properties-panel.update-businessobject', {
+          element: element,
+          businessObject: context,
+          properties: { properties: newProperties }
+        });
+      }
     }
   }
 
-  function updateProperties(element, newTemplate) {
+  function updateProperties(element, newTemplate, context) {
 
+    if (!context) {
+      context = getOrCreateExtensionElements(element);
+    }
     var newProperties = createBpmnPropertyUpdates(newTemplate, bpmnFactory);
 
     if (Object.keys(newProperties).length > 0) {
       modeling.updateProperties(element, newProperties);
     }
   }
 
-  function updateInOut(element, newTemplate) {
-
-    var extensionElements;
+  function updateInOut(element, newTemplate, context, inside) {
 
     var newInOut = createCamundaInOut(newTemplate, bpmnFactory),
         oldInOut;
 
     if (newInOut) {
-      extensionElements = getOrCreateExtensionElements(element);
-
-      oldInOut = findExtensions(extensionElements, [ 'camunda:In', 'camunda:Out' ]);
-
-      commandStack.execute('properties-panel.update-businessobject-list', {
-        element: element,
-        currentObject: extensionElements,
-        propertyName: 'values',
-        objectsToAdd: newInOut,
-        objectsToRemove: oldInOut
-      });
+      if (!context) {
+        context = getOrCreateExtensionElements(element);
+      }
+      if (!inside) {
+        oldInOut = findExtensions(context, [ 'camunda:In', 'camunda:Out' ]);
+
+        commandStack.execute('properties-panel.update-businessobject-list', {
+          element: element,
+          currentObject: context,
+          propertyName: 'values',
+          objectsToAdd: newInOut,
+          objectsToRemove: oldInOut
+        });
+      }
+      else {
+        commandStack.execute('properties-panel.update-businessobject', {
+          element: element,
+          businessObject: context,
+          properties: { inout: newInOut }
+        });
+      }
     }
   }
 
-  function updateExecutionListener(element, newTemplate) {
+  function updateExecutionListener(element, newTemplate, context, inside) {
```

Comment:
Shouldn't it, based on whether context is given or not, be obvious whether we are inside?